tests/unit/lms/services/mailchimp_test.py

```diff
@@ -13,6 +13,114 @@
+        mailchimp_client.messages.send.assert_called_once_with(
+            {
+                "message": {
+                    "subject": Any.string(),
+                    "html": Any.string(),
+                    "subaccount": sentinel.subaccount_id,
+                    "from_email": sentinel.from_email,
+                    "from_name": sentinel.from_name,
+                    "to": [{"email": sentinel.to_email, "name": sentinel.to_name}],
+                    "track_opens": True,
+                    "track_clicks": True,
+                    "auto_text": True,
+                    "headers": {},
+                },
+                "async": True,
+            }
+        )
+
+    def test_it_doesnt_send_duplicate_emails(
+        self, svc, mailchimp_client, sender, recipient
+    ):
+        # Try to send the same email twice.
+        for _ in range(2):
+            svc.send(
+                "lms:templates/email/instructor_email_digest/",
+                sender,
+                recipient,
+                template_vars={},
+                task_done_key="test_key",
+            )
+
+        # It should only have sent the email once.
+        assert len(mailchimp_client.messages.send.call_args_list) == 1
+
+    def test_with_unsubscribe_url(self, svc, mailchimp_client, sender, recipient):
+        svc.send(
+            "lms:templates/email/instructor_email_digest/",
+            sender,
+            recipient,
+            {},
+            sentinel.unsubscribe_url,
+        )
+
+        mailchimp_client.messages.send.assert_called_once_with(
+            Any.dict.containing(
+                {
+                    "message": Any.dict.containing(
+                        {"headers": {"List-Unsubscribe": sentinel.unsubscribe_url}}
+                    )
+                }
+            )
+        )
+
+    def test_if_mailchimp_client_raises(self, svc, mailchimp_client, sender, recipient):
+        original_exception = RuntimeError("Mailchimp crashed!")
+        mailchimp_client.messages.send.side_effect = original_exception
+
+        with pytest.raises(MailchimpError) as exc_info:
+            svc.send(
+                "lms:templates/email/instructor_email_digest/", sender, recipient, {}
+            )
+        assert exc_info.value.__cause__ == original_exception
+
+    @pytest.fixture
+    def mailchimp_client(self, mailchimp_transactional):
+        return mailchimp_transactional.Client.return_value
+
+    @pytest.fixture
+    def sender(self):
+        return EmailSender(
+            sentinel.subaccount_id,
+            sentinel.from_email,
+            sentinel.from_name,
+        )
+
+    @pytest.fixture
+    def recipient(self):
+        return EmailRecipient(
+            sentinel.to_email,
+            sentinel.to_name,
+        )
+
+    @pytest.fixture
+    def svc(self, db_session):
+        return MailchimpService(db_session, sentinel.api_key)
```

Comment:
This test class it just the `TestSendTemplate` class below copy-pasted and adjusted to test `send()` instead of `send_template()`